### PR TITLE
Add github pages documentation

### PR DESCRIPTION
**Pull request by kamildoleglo**
~This requires enabling Github pages for `master/docs` in repo settings, test version visible here: 
https://kamildoleglo.github.io/dokka/~
Enabled: http://kotlin.github.io/dokka

After that we should probably refactor the README and include a link to those docs

### CODE REVIEW

File: docs/src/doc/docs/user_guide/cli/usage.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  * `-cacheRoot` - cache directory to enable package-list caching
+  * `-pluginsClasspath` - artifacts with dokka plugins, separated by `;`. At least dokka base and all its dependencies must be added there 
+  * `-offlineMode` - do not resolve package-lists online
+  * `-failOnWarning` - throw an exception instead of a warning
+  * `-globalPackageOptions` - per package options added to all source sets
+  * `-globalLinks` - external documentation links added to all source sets
+  * `-globalSrcLink` - source links added to all source sets
+  * `-sourceSet` - (repeatable) - configuration for a single source set. Following this argument, you can pass other arguments:
+    * `-moduleName` - (required) - module name used as a part of source set ID when declaring dependent source sets
+    * `-moduleDisplayName` - displayed module name
```
Comment (sellmair) [inline docs/src/doc/docs/user_guide/cli/usage.md:23]: This does not exist anymore! Check recent master for this, I updated the Readme there already 👍

File: docs/src/doc/docs/user_guide/gradle/usage.md [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+    // When set to true kotlinTasks are also omitted
+    disableAutoconfiguration.set(false)
+
+    // Use default or set to custom path to cache directory
+    // to enable package-list caching
+    // When this is set to default, caches are stored in $USER_HOME/.cache/dokka
+    cacheRoot.set(file("default"))
+    dokkaSourceSets {
+        configureEach { // Or source set name, for single-platform the default source sets are `main` and `test`
+            moduleDisplayName.set("data")
```
Comment (sellmair) [inline docs/src/doc/docs/user_guide/gradle/usage.md:93]: Also removed

File: docs/src/doc/docs/user_guide/gradle/usage.md [hunk 3]
```diff
@@ -0,0 +1,293 @@
...
+The available configuration options for are shown below:
+
+```kotlin
+dokkaHtml {
+    outputDirectory.set(buildDir.resolve("dokka"))
+
+    // Used for disabling auto extraction of sources and platforms
+    // When set to true kotlinTasks are also omitted
+    disableAutoconfiguration.set(false)
+
```
Comment (sellmair) [inline docs/src/doc/docs/user_guide/gradle/usage.md:86]: We can also set moduleName now here!

File: docs/src/doc/docs/user_guide/gradle/usage.md [hunk 4]
```diff
@@ -0,0 +1,293 @@
...
+                prefix.set("kotlin.internal") // will match kotlin.internal and all sub-packages of it
+                suppress.set(true)
+            }
+        }
+    }
+```
+
+## Multiplatform
+Dokka supports single-platform and multi-platform projects using source sets abstraction. For most mutli-platform projects
+you should assume that dokka's source sets correspond to Kotlin plugin's source sets. Main source sets will be registered
```
Comment (sellmair) [inline docs/src/doc/docs/user_guide/gradle/usage.md:187]: All source sets are now automatically registered. Test source sets will just be suppressed!

File: docs/src/doc/docs/user_guide/maven/usage.md [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+            </goals>
+        </execution>
+    </executions>
+    <configuration>
+    
+        <!-- Set to true to skip dokka task, default: false -->
+        <skip>false</skip>
+    
+        <!-- Default: ${project.artifactId} -->
+        <moduleDisplayName>data</moduleDisplayName>
```
Comment (sellmair) [inline docs/src/doc/docs/user_guide/maven/usage.md:66]: Same as above ☺️

**Review by sellmair — CHANGES_REQUESTED** [review]

File: docs/src/doc/docs/user_guide/cli/usage.md (hunk 1 above)
Comment (kamildoleglo) [reply docs/src/doc/docs/user_guide/cli/usage.md:23]: Strange, took that from master 🤔 . Presumably I didn't have the most recent one, will fix

Comment (sellmair) [reply docs/src/doc/docs/user_guide/cli/usage.md:23]: Merged a few minutes ago 🙈 
🤦 Sorry

Comment (kamildoleglo) [reply docs/src/doc/docs/user_guide/cli/usage.md]: Done ✅

File: docs/src/doc/docs/user_guide/gradle/usage.md (hunk 2 above)
Comment (kamildoleglo) [reply docs/src/doc/docs/user_guide/gradle/usage.md]: Done ✅

File: docs/src/doc/docs/user_guide/gradle/usage.md (hunk 3 above)
Comment (kamildoleglo) [reply docs/src/doc/docs/user_guide/gradle/usage.md:87]: Added ✅

File: docs/src/doc/docs/user_guide/gradle/usage.md (hunk 4 above)
Comment (kamildoleglo) [reply docs/src/doc/docs/user_guide/gradle/usage.md]: Done ✅

File: docs/src/doc/docs/user_guide/maven/usage.md (hunk 5 above)
Comment (kamildoleglo) [reply docs/src/doc/docs/user_guide/maven/usage.md]: Done ✅

**Comment by kamildoleglo** [comment]
Should we also refactor the README in this PR?

**Comment by sellmair** [comment]
> Should we also refactor the README in this PR?

Yes, this would be 🔥 👍

**Review by sellmair — APPROVED** [review]

**Review by Kordyjan — APPROVED** [review]

**Comment by sellmair** [comment]
Cherry picked into 1.4.0 🍒

**Comment by dschuermann** [comment]
@kamildoleglo @sellmair Just want to thank you :) I hacked together a dokka plugin for our Hugo site: https://github.com/cotechde/dokka-hugo-plugin
And it works.

**Pull request merged**